src/lando/api/legacy/api/try_push.py

```diff
@@ -98,10 +98,8 @@ def parse_revisions_from_request(
         )
 
 
-@auth.require_auth0(scopes=("openid", "lando", "profile", "email"), userinfo=True)
-# Re-enable this check once our Auth0 instance returns group membership for access
-# tokens granted via the Device Authorization flow.
-# @auth.enforce_request_scm_level(SCM_LEVEL_1)
+@require_authenticated_user
+@require_permission("scm_level_1")
```

Comment:
Tokens granted via Device Authorization Flow don't include scm levels, so we can't enforce the permissions here. Please re-add the comment from the original.

Reply:
> Tokens granted via Device Authorization Flow don't include scm levels, so we can't enforce the permissions here. Please re-add the comment from the original.

So as implemented in this PR, this endpoint will require a user to be authenticated via Django's auth framework, and thus we can directly check whether the permission is there or not. This endpoint will not work as it did previously (e.g., via an auth token). We should file a bug to implement this change in Mach (and Lando, since we don't currently have a way to authenticate with Lando via CLI). I will add a comment and file a bug for this.

Reply:
Please describe the change you think we would make here in more detail. How will we implement authentication to Lando in Mach without using an access token granted from Auth0? The statement "we don't currently have a way to authenticate with Lando via CLI" is not accurate, we can authenticate to Lando via the CLI, but the granted access token doesn't include the `https://sso.mozilla.com/claim/groups` key with group membership information.

Reply:
> The statement "we don't currently have a way to authenticate with Lando via CLI" is not accurate

My statement is referring to new Lando, not legacy Lando -- should have been less ambiguous about that.

> How will we implement authentication to Lando in Mach without using an access token granted from Auth0?

This is something yet to be determined. We could do it via Auth0 still, but ultimately we would have to authenticate with Django so that all the requests get tied to the Django user. Regardless, the commented out portion here would be irrelevant since Django can check the permissions directly.

Reply:
It sounds like you're suggesting we roll our own auth system for Lando. Without more details, it sounds like this is far outside the scope of the Django migration. Keep in mind that while we are trying to leverage the Django permissions system, Auth0 is still the system which provides information about the users and their permissions (including important things like removing/disabling users when they leave the company). I don't think we can or should be implementing a form of auth that doesn't use Auth0 access tokens.

Reply:
> It sounds like you're suggesting we roll our own auth system for Lando.

Not sure what you mean here? Perhaps you can elaborate. Since the endpoint is a Django one, therefore Django will have to determine if a user is authorized or not.

The new Django-based auth system is already part of the scope of this Lando porting, so is improving/"modernizing" the auth story in Lando. The way we are initially using it is via an Auth0 backend to authenticate users, and to parse user permissions via userinfo.

After chatting with @globau we've decided that the particular auth implementation for this particular endpoint (try) can be deferred to after the initial launch. One possibility is to use the Auth0 token to authenticate the user with Django directly via the endpoint (and create the user/permissions at that point), but there are other possible approaches as well that we can discuss.

Reply:
Filed [bug 1909723](https://bugzilla.mozilla.org/show_bug.cgi?id=1909723).

Reply:
> > It sounds like you're suggesting we roll our own auth system for Lando.
> 
> Not sure what you mean here? Perhaps you can elaborate. Since the endpoint is a Django one, therefore Django will have to determine if a user is authorized or not.

You previously stated the "endpoint will not work as it did previously (e.g., via an auth token)". Since you haven't given any more details ("This is something yet to be determined"), it sounds like you are intending to implement some non-Auth0 access token authentication to Lando.

Reply:
> it sounds like you are intending to implement some non-Auth0 access token authentication to Lando.

That's still a possibility (i.e., authenticating directly with Lando via the CLI, or a web page, or in some other way). Ultimately Lando will have a user record for each user who authenticates one way or another, and those users as long as they exist in Auth0 can be tied together regardless of how they originate in the system (e.g., via Auth0 token or via a web login). In any case, one simple approach (if indeed possible) would be to use Lando to fill in the missing information after validating the token that is generated with the current approach in Mach.

Reply:
> "endpoint will not work as it did previously (e.g., via an auth token)"

Lastly to clarify this point, this is correct. With this implementation, the endpoint does not work with the auth token (in fact, no endpoint in Django supports an auth token at this point).